ui/jobs/buff_tracker.ts

```diff
@@ -445,99 +519,106 @@ export class BuffTracker {
     };
 
-    for (let i = 0; i < keys.length; ++i) {
-      const buff = this.buffInfo[keys[i]];
-      buff.name = keys[i];
+    for (const key in this.buffInfo) {
+      const buff = this.buffInfo[key];
+      if (!buff) continue;
+      buff.name = key;
 
-      const overrides = this.options.PerBuffOptions[buff.name] || {};
-      buff.borderColor = overrides.borderColor || buff.borderColor;
-      buff.icon = overrides.icon || buff.icon;
-      buff.side = overrides.side || buff.side || 'right';
-      buff.sortKey = overrides.sortKey || buff.sortKey;
-      buff.hide = overrides.hide === undefined ? buff.hide : overrides.hide;
+      const overrides = this.options.PerBuffOptions[buff.name] ?? null;
+      buff.borderColor = overrides?.borderColor ?? buff.borderColor;
+      buff.icon = overrides?.icon ?? buff.icon;
+      buff.side = overrides?.side ?? buff.side ?? 'right';
+      buff.sortKey = overrides?.sortKey || buff.sortKey;
+      buff.hide = overrides?.hide ?? buff.hide ?? false;
 
       for (const prop in propToMapMap) {
         if (!(prop in buff))
           continue;
-        const key = buff[prop];
+        const key = buff[prop as keyof typeof propToMapMap];
         if (typeof key === 'undefined') {
           console.error('undefined value for key ' + prop + ' for buff ' + buff.name);
           continue;
         }
 
-        const map = propToMapMap[prop];
+        const map = propToMapMap[prop as keyof typeof propToMapMap];
         map[key] = map[key] || [];
-        map[key].push(buff);
+        map[key]?.push(buff);
       }
     }
 
-    const v520 = {
-      // identical with latest patch
-      /* example
-      trick: {
-        durationSeconds: 10,
-      },
-      */
-    };
-
-    const buffOverrides = {
-      cn: v520,
-      ko: v520,
-    };
-
-    for (const key in buffOverrides[this.options.ParserLanguage]) {
-      for (const key2 in buffOverrides[this.options.ParserLanguage][key])
-        this.buffInfo[key][key2] = buffOverrides[this.options.ParserLanguage][key][key2];
-    }
+    // const v520 = {
+    //   // identical with latest patch
+    //   /* example
+    //   trick: {
+    //     durationSeconds: 10,
+    //   },
+    //   */
+    // };
+
+    // const buffOverrides = {
+    //   cn: v520,
+    //   ko: v520,
+    // };
+
+    // for (const key in buffOverrides[this.options.ParserLanguage]) {
+    //   for (const key2 in buffOverrides[this.options.ParserLanguage][key])
+    //     this.buffInfo[key][key2] = buffOverrides[this.options.ParserLanguage][key][key2];
+    // }
   }
 
-  onUseAbility(id, matches) {
+  onUseAbility(id: string, matches: Matches<BaseRegExp<AbilityParams>>): void {
     const buffs = this.gainAbilityMap[id];
     if (!buffs)
       return;
 
     for (const b of buffs)
-      this.onBigBuff(b.name, b.durationSeconds, b, matches.source);
+      this.onBigBuff(b.name ?? '', b.durationSeconds ?? 0, b, matches?.source ?? '');
```

Comment:
I think this would be cleaner if these were optional parameters so that the function itself can handle it rather than the caller.  (Also, are all these fields really optional on buffs?)

Reply:
Changed as default value.